specifyweb/frontend/js_src/lib/components/TreeView/index.tsx

```diff
@@ -104,14 +104,31 @@ function TreeView<SCHEMA extends AnyTree>({
   );
 
   // FEATURE: synchronize focus path with the URL
-  const [focusPath = [], setFocusPath] = useCachedState(
+  const [_focusedPath = [], setFocusPath] = useCachedState(
     'tree',
     `focusPath${tableName}`
   );
+  const [focusPath1 = [], setFocusPath1] = React.useState<RA<number>>();
+  const [focusPath2 = [], setFocusPath2] = React.useState<RA<number>>();
 
   const [focusedRow, setFocusedRow] = React.useState<Row | undefined>(
     undefined
   );
+  const [focusedRow1, setFocusedRow1] = React.useState<Row | undefined>(
+    undefined
+  );
+  const [focusedRow2, setFocusedRow2] = React.useState<Row | undefined>(
+    undefined
+  );
+
+  const [lastFocusedTree, setLastFocusedTree] = React.useState<boolean>(true);
+
+  React.useEffect(() => {
+    if (lastFocusedTree) setFocusedRow(focusedRow1);
+    else {
+      setFocusedRow(focusedRow2);
+    }
+  }, [focusedRow1, focusedRow2]);
```

Comment:
This doesn't need to be a useEffect. Simplify:

```
const focusedRow = lastFocusedTree ? focusedRow1 : focusedRow2;
```

and then focusedRow is not a state, but just a simple variable